src/charm.py

```diff
@@ -52,18 +52,39 @@ def __init__(self, *args: Any) -> None:
             host=f"{self.app.name}-endpoints.{self.model.name}.svc.cluster.local",
             strip_prefix=True,
         )
+        self.framework.observe(self.on.config_changed, self._on_config_changed)
+        self.framework.observe(self.on.reset_instance_action, self._reset_instance_action)
```

Comment:
Should this be `self._on_reset_instance_action`?

Reply:
Good catch, completely forgot. Changed.